extensions/amp-analytics/0.1/events.js

```diff
@@ -1474,61 +1479,104 @@ export class VisibilityTracker extends EventTracker {
       );
     }
 
-    let unlistenPromise;
     // Root selectors are delegated to analytics roots.
     if (!selector || selector == ':root' || selector == ':host') {
       // When `selector` is specified, we always use "ini-load" signal as
       // a "ready" signal.
-      unlistenPromise = visibilityManagerPromise.then(
-        visibilityManager => {
-          return visibilityManager.listenRoot(
-            visibilitySpec,
-            this.getReadyPromise(waitForSpec, selector),
-            createReportReadyPromiseFunc,
-            this.onEvent_.bind(
-              this,
-              eventType,
-              listener,
-              this.root.getRootElement()
-            )
-          );
-        },
-        () => {}
+      unlistenPromises.push(
+        visibilityManagerPromise.then(
+          visibilityManager => {
+            return visibilityManager.listenRoot(
+              visibilitySpec,
+              this.getReadyPromise(waitForSpec, selector),
+              createReportReadyPromiseFunc,
+              this.onEvent_.bind(
+                this,
+                eventType,
+                listener,
+                this.root.getRootElement()
+              )
+            );
+          },
+          () => {}
+        )
       );
     } else {
       // An AMP-element. Wait for DOM to be fully parsed to avoid
       // false missed searches.
       const selectionMethod =
         config['selectionMethod'] || visibilitySpec['selectionMethod'];
-      unlistenPromise = this.root
-        .getAmpElement(
-          context.parentElement || context,
-          selector,
-          selectionMethod
-        )
-        .then(element => {
-          return visibilityManagerPromise.then(
-            visibilityManager => {
-              return visibilityManager.listenElement(
+      selector = Array.isArray(selector) ? selector : [selector];
+      // Array selectors do not suppor the special cases: ':host' & ':root'
+      for (let i = 0; i < selector.length; i++) {
+        unlistenPromises.push(
+          this.root
+            .getAmpElement(
+              context.parentElement || context,
+              selector[i],
+              selectionMethod,
+              multiSelectorVisibilityOn
+            )
+            .then(element =>
+              this.getUnlistenPromiseForElement_(
```

Comment:
Since `getUnlistenPromiseForElement_` is only used once here, shall we inline `visiblityManagerPromise.then()....`?

Reply:
Yup good point

Reply:
[````
```]()